src/main/java/bio/terra/app/controller/DuosApiController.java

```diff
@@ -0,0 +1,83 @@
+package bio.terra.app.controller;
+
+import bio.terra.app.configuration.ApplicationConfiguration;
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.common.iam.AuthenticatedUserRequestFactory;
+import bio.terra.controller.DuosApi;
+import bio.terra.model.DuosFirecloudGroupModel;
+import bio.terra.service.auth.iam.IamAction;
+import bio.terra.service.auth.iam.IamResourceType;
+import bio.terra.service.auth.iam.IamService;
+import bio.terra.service.duos.DuosService;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.swagger.annotations.Api;
+import java.util.List;
+import java.util.Optional;
+import javax.servlet.http.HttpServletRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+
+@Controller
+@Api(tags = {"duos"})
+public class DuosApiController implements DuosApi {
+
+  private final Logger logger = LoggerFactory.getLogger(DuosApiController.class);
+
+  private final HttpServletRequest request;
+  private final ObjectMapper objectMapper;
+  private final AuthenticatedUserRequestFactory authenticatedUserRequestFactory;
+  private final ApplicationConfiguration appConfig;
+  private final IamService iamService;
+  private final DuosService duosService;
+
+  public DuosApiController(
+      HttpServletRequest request,
+      ObjectMapper objectMapper,
+      AuthenticatedUserRequestFactory authenticatedUserRequestFactory,
+      ApplicationConfiguration appConfig,
+      IamService iamService,
+      DuosService duosService) {
+    this.request = request;
+    this.objectMapper = objectMapper;
+    this.authenticatedUserRequestFactory = authenticatedUserRequestFactory;
+    this.appConfig = appConfig;
+    this.iamService = iamService;
+    this.duosService = duosService;
+  }
+
+  @Override
+  public ResponseEntity<List<DuosFirecloudGroupModel>> retrieveDuosFirecloudGroups() {
+    return ResponseEntity.ok(duosService.retrieveFirecloudGroups());
+  }
+
+  @Override
+  public ResponseEntity<DuosFirecloudGroupModel> retrieveDuosFirecloudGroup(String duosId) {
+    return ResponseEntity.ok(duosService.retrieveFirecloudGroup(duosId));
```

Comment:
Ah, I'm glad you mentioned this because it's not that common a pattern in TDR.  We want to ensure that the user is logged in but don't have a specific permission to check, which is why I didn't make it an unauthed endpoint… the root object here is a DUOS dataset's metadata which is accessible to logged in users in DUOS.  DUOS datasets aren't resources in Sam.